.github/workflows/release-step-3.yml

```diff
@@ -200,11 +200,35 @@ jobs:
           group-id: 'co.elastic.otel'
           artifact-id: 'elastic-otel-javaagent'
           version: ${{ inputs.version }}
+  
+  generate-release-notes:
+    runs-on: ubuntu-latest
+    name: Generate the changelog for this release
+    outputs:
+      notes: ${{ steps.print_release_notes.outputs.notes }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.ref }}
+      - name: Setup Gradle
+        uses: ./.github/workflows/gradle-goal
+        with:
+          command: ""
+      - name: Print Release Notes
+        id: print_release_notes
+        run: |
+          echo 'notes<<RELNOTESEOF' >> $GITHUB_OUTPUT
+          cat CHANGELOG.next-release.md >> $GITHUB_OUTPUT
+          printf '\nThis release is based on the following upstream versions:\n\n' >> $GITHUB_OUTPUT
+          ./gradlew -q printUpstreamDependenciesMarkdown >> $GITHUB_OUTPUT
+          echo 'RELNOTESEOF' >> $GITHUB_OUTPUT  
```

Comment:
[minor] could it be simpler to implement this completely as a gradle task ?

Reply:
I don't think it would be really simpler because then the gradle task would have to do the file handling, which is more code than `cat CHANGELOG.next-release.md`.
The multiline support code `echo 'notes<<RELNOTESEOF' >> $GITHUB_OUTPUT` and `echo 'RELNOTESEOF' >> $GITHUB_OUTPUT ` would still be required, so we wouldn't save much